tests/test_product_manager.py

```diff
@@ -0,0 +1,104 @@
+
+from . import setup_test
+import unittest
+from moto import mock_s3
+import database_connector
+import database_manager
+import product_manager
+from image_manager import ImageManager
+from image_manager import AWSConnector
+
+
+@mock_s3
+class TestProductManager(unittest.TestCase):
+    bucket = 'jf-test555-bucket'
+
+    @classmethod
+    def setUpClass(self):
+        self.database_connector = database_connector.DatabaseConnector()
+        self.item_manager = database_manager.ItemManager(
+            self.database_connector)
+        self.tag_manager = database_manager.TagManager(
+            self.database_connector)
+        aws_connector = AWSConnector()
+        self.s3_resource = aws_connector.get_s3_resource()
+        self.image_manager = ImageManager(self.s3_resource, self.bucket)
+        self.product_manager = product_manager.ProductManager(
+            self.item_manager, self.tag_manager, self.image_manager)
+
+    def setUp(self):
+        setup_test.create_test_database_setup(self.database_connector)
+        self.s3_resource.create_bucket(Bucket=self.bucket, CreateBucketConfiguration={
+            'LocationConstraint': 'eu-west-2'},)
+
+    def test_add_product(self):
+        # arrange
+        tags = ['bird', 'fineline']
```

Comment:
Setting this to a tuple has two advantages:

1. It makes sure the code under test does not try to modify it.
2. Your test can't accidentally pass if this is accidentally modified.

Reply:
updated